tests/integration/test_gerrit.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+
+import logging
+
+#################################################################################
+# Enable logging
+
+log = logging.getLogger('test.gerrit')
+
+#################################################################################
+
+from tests.helpers import GitRepoTestCase
+
+from git_repo.services.service import gerrit
+
+class Test_Gerrit(GitRepoTestCase):
+    log = log
+
+    def get_service(self):
+        return gerrit.GerritService(c={
+            '__name__': 'gitrepo "gerrithub"',
+            'fqdn': 'review.gerrithub.io',
+            'username': 'TestUser',
+            'token': 'test_token'
+        })
+
+    def get_requests_session(self):
+        return self.service.session
+
+    def action_request_create(self, namespace, repository, branch):
```

Comment:
because `action_request_create` is quite different from the others, maybe it's a good thing to overload it. The only thing that bugs me is that we lose the factorisation it offers in `helpers.py`, so that with a change alike the one you made going from return to yield, you change it once for every service, and once for all the tests.

Maybe a good idea could be to simply not overload `action_request_create` in the service, but instead move that method to `helpers.py`, name it `action_request_create_by_push` and leave it alongside `action_request_create`.